### DISCUSSION TITLE
[Feature] allow use of explore features for shared albums

### DISCUSSION THREAD

**Discussion by crutcha** [immich-app/immich#3176] [Feature Request]
### The feature

Currently when albums are shared to a user, a user can only see what's shared with them via the `Sharing` pane in the UI. It would be nice if shared photos/albums were treated more like your own photos. Mainly, this means identified persons in shared photos show up under the `Explore` pane so users can filter by person in photos that are shared with them, but this could also entail these photos showing up in the main `Photos` pane timeline view. 

### Platform

- [X] Server
- [X] Web
- [ ] Mobile

**Comment by CDrummond** [comment]
Likewise, location information does not seem to be used for partner images.

Currently the 'Explore/Places' view only contains locations from images of the current user. There is, AFAICT, no way to currently use locations from partner images. It would be useful if the 'Explore/Places' view combined all locations, so that showing images from a location will show mine and any partner images. Searching on partner locations would also be beneficial.

When finding the partner feature I hoped it basically allowed full (read-only) access to their library - so albums, locations, etc.

**Reply by schmitzkr** [reply]
Is this the same as this request?

https://github.com/immich-app/immich/discussions/2846

**Reply by jtlarson** [reply]
I think they are complimentary, but not the same. 

#2846 focuses on adding images shared by partners to the main view, whereas this request is for sharing of data (recognized faces, locations, etc) in the "Explore" view and search.

**Reply by dakabali** [reply]
Similar to the Places the recognized Faces of each shared photo should also be shared. The target user should be able to choose to manage the shared faces - for example merge them with exisiting recognized faces of own photos. Actually all shared photos (albums) should be transparent with the available information for the partner / user who's shared with - equally like own photos.

**Reply by jrasm91** [reply]
FWIW - we have at least added an implementation for partner sharing, which we did not use to have. It works exactly like Google photos does - separate view specifically for those photos. In Google photos partner photos don't show up on your timeline (unless you opt to copy every image into your own library, which is irreversible), and afaik you cannot search them directly.

There are plans to have better integration of shared assets in the future, but it is a work in progress. The backlog is quite long. I think having the ability to view the photos of a good first step and tighter integrations will file in the future. Specifically, I think it makes sense to optionally view them in your timeline, have them included in search results, and include them on the map view.

**Reply by schmitzkr** [reply]
I agree that the sharing functionality is very useful as is (and fully appreciate work piling up in the growing backlog).  Thanks!

In the past while using pcloud, I had to use one account between my partner and I.  This is a useful workaround, everything is appears as one user and can easily be seen by both. We could use metadata to recall which device an image came from. I may resort to this with immich.

My biggest need here is to be able to select a group of assets and share them with my kids grandparents.  Right now with two users, sometimes I am the owner of the assets and other times it's my partner.
To share these assets I need to do two shares, and sometimes I can't recall which shots I shared from the first batch. It is also confusing for the grandparents ("why are you sharing twice") Slightly compounding things is the inability to multi select partner assets on android, meaning I need to do them one at a time or get to webui to move things into an album and share that.

I have been off Google photos for a few years now, and so maybe I'm not remembering correctly, but I don't recall these pain points.  Hopefully my case is helpful at some point.
Love immich so far and am amazed by the speed of development and the level of support!
Thanks again.

**Reply by jrasm91** [reply]
Yeah, that is a perfect use case IMO. In google everything eventually is a form of an album and both users can contribute photos to it. You can make a shared link for an album in Immich, so that would be one option - create an album with the first batch, add assets from the second user, and then one of the two could create a share link for the album.

**Comment by ThisIsTheOnlyUsernameAvailable** [comment]
I think this has some similarities to my suggestion.  I'll leave a link in case either gets considered, the other may influence how it works.

https://github.com/immich-app/immich/discussions/2846

**Comment by tuta23** [comment]
I think this request is the proper place to add to -- for my money, I think being able to favorite shared photos would be nice.  

Additionally, seeing Shared photos by categories (Videos, motion, panoramic) would also be greatly appreciated.

**Comment by datmospro** [comment]
UP! It must be an option. Share albums must be able to identify people.

**Comment by AxeC100** [comment]
Yeah this is absolutely huge. Right now if me and my partner just want to see a consolidated view of all the photos of let's say myself across both our libraries, there isn't a way to do that easily. I'd have to manually create a shared album with every picture of myself in both libraries and then just look in the album, which is much more hassle than just being able to use the people filter to instantly view all of these.

Worth noting based on the above that it shouldn't necessarily just apply to shared albums, if your partner sharing it should probably apply to the whole 'joint timeline'.

**Reply by AndyPro720** [reply]
You do have partner view if all you wish to simply do is share all photos with each other. You can look the feature up!

**Reply by dakky** [reply]
There is a slight difference between "share everything" and "share some albums with all related features" imho :smiley:

**Comment by marcelo-mbwr** [comment]
**I am absolutely with you here. In general, I also think that there should be the possibility of using the media from shared albums to give the users with whom it was shared the feeling that the photos also belong to them.**

_For example, I share the most albums with the family because I always take a lot of photos on vacation. Unfortunately, my family members don't have the pictures in their timeline on the home page and other cool Immich features are not available unless they download the pictures themselves. That's too bad._

Maybe an option in the settings would be great to say that you always want to download all pictures from shared albums. This would create quite an overhead because the images in an album would then be saved several times for different users, but the images would then be physically assigned to each person.

But perhaps a virtual implementation also stands in the focus, so that media from shared albums could be treated as if it were its own medium. Of course, it's a shame in this case that if you theoretically want to use the media independently of Immich someday, the images are not owned by the respective users and were only saved by the person who uploaded them.

So the first option sounds much more attractive to me. I hope the developers gain insight from this discussion. And as always: Thanks to the Immich team - it's the best software in a long time.

_@crutcha Feel free to include this generic description in the feature request if you wish._

**Reply by Nassiel** [reply]
It's not doing a "copy", IMHO, it's about sharing the actual metadata with the picture. If I share the image with you, I want you to know who is there plus where/when it is being taken because the whole thing is the picture processing.

To share the images (jpg) and have a gallery, there are plenty of other more straightforward software like Nextcloud, which this part works perfectly.

_**In one of the current use cases, my father took ALL the family pictures, digitalized them, uploaded them, corrected all the metadata (faces, locations and comments) to share with the rest of the family.**_

It would be useless, to just download the gallery and get the images, which is the interesting the knowledge of when, where, who is there and some personal comments like "we were visiting our uncle Peter".

**Reply by DrSlump74** [reply]
I agree that shared albums should also share metadata and these should be inserted in the central database to allow a search, making it seem like these shared photos are also my property.
But there is a problem, if for some reason the album is removed from sharing, it is as if you had deleted it, but it was not you but the person who shared it with you in the first place.
This would be fine, but I would also like to have an additional choice, the possibility of downloading the shared album to my library.
I am aware that by definition a shared album is a mere link to the resource and does not necessarily have to be definitive, but I would also like to have the possibility of deciding it, I who receive the link, obviously if the person who creates the link allows it. So an additional option that allows me to download the album's photos to my library could be useful, obviously it would invalidate the concept of a shared album.

**Comment by DrSlump74** [comment]
Hi, i'll explain my experience with Google Photos for this feature.
I take photos, put them in an album and share them with my wife, she can view them and do what she wants with them, but they are still photos shared via album.
Google Photos adds the feature to download the shared album, this way my wife also has the same album in her archive and therefore it will be managed in the timeline and in searches between places, faces, etc...

**Comment by AndyPro720** [comment]
Hey @alextran1502 , will we see this, or a workaround for this to be scheduled in Immich's projected releases/roadmap?

**Comment by AndyPro720** [comment]


**Reply by bo0tzz** [reply]
@AndyPro720 please don't ping people for updates. Also see https://justinmayer.com/posts/any-updates/.

**Comment by Yecgaa1** [comment]
I think this feature is valuable, after all it's a bit frustrating to share albums but not have a way to explore everyone's story
